### DISCUSSION TITLE
Maximum call stack size exceeded When recurrsion

### DISCUSSION THREAD

**Discussion by koki0702** [pyodide/pyodide#2818] [Q&A]
I got runtime error when I called recursion function(min-max algorithm).

Is there any way to know the maximum size of the call stack?
(Or change the maximum size of the call stack?)

<img width="429" alt="image" src="https://user-images.githubusercontent.com/515809/176824538-90dc35d1-d5a3-4f23-b8bb-50106b29eee3.png">

**Comment by rth** [comment]
Could you please share your browser and OS versions? 

For measuring the maximum recursion depth, see the code [here](https://github.com/pyodide/pyodide/blob/main/benchmark/stack_usage.py#L35)
Currently, this script would return,
```
browser  py_limit  py_usage  js_depth  py_depth  js_depth/py_usage  js_depth/py_depth
firefox      1000    139.70     39103      1996                279                 19
chrome       1000     11.34     13950      1225               1230                 11
node         1000     11.38     15424      1349               1355                 11
```
so the maximum recursion depth should be 1000 (the default) for Chrome and Firefox on laptop/desktop. Though the stack size is mostly specified by the browser, so it might be lower on mobile or in other browsers.

**Comment by koki0702** [comment]
> maximum recursion depth should be 1000

Thanks for answering!
I just fixed my code. And the code worked well.

BTW, I created tic-tac-toe game (vs. AI player) using PyScript.
PyScript is awesome! 

https://koki0702.github.io/tictactoe-pyscript/


![tic](https://user-images.githubusercontent.com/515809/176858570-1199b7a0-8fa2-44be-8788-8eef539a9bdc.gif)
